insert_string_tpl.h

```diff
@@ -68,7 +68,7 @@ Z_INTERNAL void INSERT_STRING(deflate_state *const s, const uint32_t str, uint32
         uint32_t val, hm, h = 0;
 
 #ifdef UNALIGNED_OK
```

Comment:
Instead of checking if UNALIGNED_OK is defined, we would need to check if system is little-endian or big-endian.